### ISSUE TITLE
How to reverse DOM to original ?

### ISSUE THREAD

**Issue by l2aelba** [verlok/vanilla-lazyload#509]
How to reverse to original HTML? (DOM)

Example

**Original DOM** (HTML)

```html
<div class="lazy" data-bg="test.jpg"></div>
```

and when we run:

```js
const lazyLoadInstance = new LazyLoad()
```

so the **output** will be something like this right?

```html
<div class="lazy loaded" data-ll-status="loaded" style="background-image: url('test.jpg');"></div>
```

But how can I **get back** to the **original HTML**?:

```html
<div class="lazy" data-bg="test.jpg"></div>
```

I tried `lazyLoadInstance.destroy()`, and it does not work :)

Thanks.

**Comment by verlok** [comment]
Hi @l2aelba,
I'm sorry I must have missed this issue back in February.

You're right, `destroy` just destroys the instance of LazyLoad, but it doesn't reverse the HTML to the original version.

It's quite unusual to want to revers the HTML to its original version.
Can you explain me what are you trying to achieve?

**Comment by l2aelba** [comment]
It for my custom CMS, I want to reverse it back and save the original HTML to db.

Thanks.

**Comment by evenreven** [comment]
This would be useful for Turbo(links) users too, since the convention there is to undo/destroy changes to DOM before navigating to a new page. The idea is to keep state on the server or in data attributes. Using vanilla-lazyload with placeholders and transitions breaks Turbo previews. (There _might_ be a way to keep vanilla-lazyload DOM changes when you revisit a page with Turbo, but I haven't got it to work.)

**Comment by verlok** [comment]
Hey guys,

sorry for the very late reply, it’s been an intense summer over here. 

Yes, I could work on introducing a new `reset` method, both as a static method to run on single images, like `LazyLoad.reset(img)` and as an instance method to reset all the images affected by `elements_selector`, to use like `lazyLoadInstance.resetAll()`. What do you think?

**Comment by verlok** [comment]
Still didn’t start working on this. I will try and find the time by this week. A [coffee injection](https://ko-fi.com/verlok) might help. 😅

Open-source software is great for everyone, but it takes passion and time to grow and evolve.

By all means, I will update you soon guys.

**Comment by verlok** [comment]
Working on it in the [feature/reset-dom](https://github.com/verlok/vanilla-lazyload/tree/feature/reset-dom) branch.

**Comment by evenreven** [comment]
Wonderful! Let me know if you need a beta tester!

**Comment by verlok** [comment]
@l2aelba, @evenreven, would you be able to try the files in the `dist` folder of the [feature/reset-dom](https://github.com/verlok/vanilla-lazyload/tree/feature/reset-dom) branch?

The usage is: 

```js
const myLL = new LazyLoad({ /*options*/ });
myLL.restoreAll();
```

NOTE: `restoreAll` brings all DOM elements controlled by LazyLoad back to their original state, but it doesn't destroy LazyLoad, so you probably want to use it along with `destroy` on TurboLinks. Like this:


```js
const myLL = new LazyLoad({ /*options*/ });
myLL.restoreAll();
myLL.destroy();
```

You also probably want to fire `restoreAll` and `destroy` when `myLL.loadingCount` is `0`, to avoid loading callbacks being called after you restore all your DOM. 

This is not your case, but it's worth mentioning that when not `destroy`ing the instance, the `update()` method could be called to restart loading restored images.

Let me know how it works for you!

**Comment by evenreven** [comment]
Looks great! I'll test and get back to you in a day or two.

**Comment by l2aelba** [comment]
Thanks! Great work!. I will try.

**Comment by verlok** [comment]
Everything is ready, code is covered by tests, and I also added a demo in the demo folders, named `restore_destroy.html`. 
I've even written the documentation in the README.

I'll just wait for your feedback to release it on the main branch.

**Comment by verlok** [comment]
I've closed the feature branch now, find everything under [the development branch](https://github.com/verlok/vanilla-lazyload/tree/development).

**Comment by verlok** [comment]
And now it's released in version 17.5.0 :)

**Comment by verlok** [comment]
Hey @l2aelba, @evenreven, 
I'm closing this issue now, since I've now released the feature you asked for.

If you’re happy with my support, the documentation and the effort I’ve been putting on this project in the latest years, I hope you might want to [buy me a coffee](https://ko-fi.com/verlok) to show your appreciation. Or [sponsor me](https://github.com/sponsors/verlok), if you're a fan.

Open-source software is great for everyone, but it takes passion and time to grow and evolve.

Thank you for thinking about it. 
Have a wonderful day!

**Comment by evenreven** [comment]
Sorry for the lack of input. The setInterval method did not work for my use case, so I tried hunting for bugs to see if it was my codebase or not before replying, but I got derailed by other projects needing my attention. Still haven't found out why setInterval doesn't work, but it works well without it, so it's still an improvement and cleaner than it used to be (I can remove the duplicate data attrib from my templates, for instance). I hope to investigate more, but the `restoreAll()` method works well, so it can wait. Thanks for the release!
